### ISSUE TITLE
state url typed param handle error

### ISSUE THREAD

**Issue by vlatkodrljin** [angular-ui/ui-router#2534]
I have following (dummy) $urlMatcherFactoryProvider type defined for one typed parameter (only for testing purposes and simulation it is false):

```
$urlMatcherFactoryProvider.type("check", {
encode: angular.identity,
decode: angular.identity,
is: function(item) {
    console.log("trigger is");
    return false;
}
```

});
and for example state defined like this:

```
$stateProvider
.state("newstate", {
     url: "/{str:check}",
     controller: function($scope) {}
})
```

What is the proper way to catch "simulated" error defined on the top, because route is not correct right?

$stateChangeError event and otherwise route in that case isn't triggered, and I think that "is" callback function is triggered multiple times and code execution after it isn't propagated below.

**Comment by Pyrolistical** [comment]
I am running into this as well.  I don't understand where a validation failure is suppose to go

**Comment by Pyrolistical** [comment]
this is the issue: https://github.com/angular-ui/ui-router/blob/0.2.18/src/state.js#L198

$state.transitionTo returns a promise which is rejected when validation fails

**Comment by christopherthielen** [comment]
This was addressed in 1.0 via db71947 param validation

End users can catch the rejection using the promise.  In addition, any `transitionTo` errors are now handled by the `defaultErrorHandler`, which logs to the console by default.
